libsolidity/ast/Types.cpp

```diff
@@ -561,6 +812,75 @@ shared_ptr<IntegerType const> IntegerConstantType::integerType() const
 		);
 }
 
+shared_ptr<FixedPointType const> RationalNumberType::fixedPointType() const
+{
+	//do calculations up here
```

Comment:
one I fixed it up it appears to be working...still gotta figure out why this infinite loop is happening though.